File: basics/17-second-largest/Solution.cpp
#include <vector>
#include <set>

int secondLargest(const std::vector<int>& arr) {
    std::set<int, std::greater<int>> s(arr.begin(), arr.end());
    if (s.size() < 2) return *s.begin();
    auto it = s.begin();
    ++it;
    return *it;
}

File: basics/17-second-largest/Solution.java
import java.util.*;

class Solution {
    public int secondLargest(int[] arr) {
        TreeSet<Integer> set = new TreeSet<>(Collections.reverseOrder());
        for (int num : arr) set.add(num);
        if (set.size() < 2) return set.first();
        set.pollFirst();
        return set.first();
    }
}
